### PR TITLE
Add simulated block devices to SimEngine implementation.

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
@agrover, @trgill  review?

File: src/sim_engine/mod.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+pub mod engine;
```
Comment (agrover) [inline src/sim_engine/mod.rs]: I think you can also re-export stuff here, so in main.rs it could still say `use sim_engine::SimEngine` if desired.

**Comment by agrover** [comment]
Also, this PR adds SimDev but also significantly puts sim_engine in its own directory, which I would probably say deserves mention in the PR title.

**Comment by mulkieran** [comment]
@agrover, @trgill review?

**Pull request merged**